docs/network-troubleshooting.adoc

```diff
@@ -0,0 +1,143 @@
+:toc: macro
+
+= Network troubleshooting
+
+:icons: font
+:numbered:
+toc::[]
+
+Clients forming the Keep Network are deployed in numerous ways using different
+infrastructure configurations and network topologies. This fact combined with
+the complexity of a p2p network may cause `keep-ecdsa` clients to experience
+some network problems occasionally. Those issues are typically noticed during
+key generation or signing processes which doesn't succeed at the first time
+and are retried. Usually, the problem is caused by clients going offline
+temporarily due to restart or network breaks. In that case, clients should
+recover automatically as `keep-ecdsa` clients are resilient and can
+continue their work after reconnecting to the network or coming up after the
+restart. However, sometimes the problem persists and troubleshooting process
+should be run on all clients taking part in the unlucky protocol execution.
+
+== Common problems
+
+Typically, there are several types of network-related problems reported by
+operators. Here is a brief description of them.
```

Comment:
It reads to me that usually there is more than one problem behind a particular key generation/signing failure. Maybe we can rephrase it into something like:

> Here are the most common problems reported by operators with a brief description for each one of them.

?

Reply:
Sure. See https://github.com/keep-network/keep-ecdsa/pull/838/commits/8437b4e2934bcc70f8a88d7b09bd1677234535ab